In [2]:
import numpy as np
import pandas as pd
import statistics as s
import matplotlib.pyplot as plt


# 1) FIND OUT HOW MANY MALES AND FEMALES PARTICIPATED IN THE TEST


In [3]:
data=pd.read_csv('C:\DSA\StudentsPerformance.csv')

In [4]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [5]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

In [6]:
data['gender'].value_counts(normalize=True)

female    0.518
male      0.482
Name: gender, dtype: float64

There are 518 females(51.8% of the total gender) and 482(48.2% of the total gender) males participated in the test.

# 2) WHAT DO YOU THINK ABOUT THE STUDENT'S PARENTAL LEVEL OF EDUCATION

In [7]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

In [36]:
data['parental level of education'].value_counts(normalize=True)

some college          0.226
associate's degree    0.222
high school           0.196
some high school      0.179
bachelor's degree     0.118
master's degree       0.059
Name: parental level of education, dtype: float64

In [8]:
data[['gender','parental level of education']].value_counts()

gender  parental level of education
female  some college                   118
        associate's degree             116
male    some college                   108
        associate's degree             106
        high school                    102
female  high school                     94
        some high school                91
male    some high school                88
female  bachelor's degree               63
male    bachelor's degree               55
female  master's degree                 36
male    master's degree                 23
dtype: int64

In [9]:
data[['gender','parental level of education']].value_counts(normalize=True)

gender  parental level of education
female  some college                   0.118
        associate's degree             0.116
male    some college                   0.108
        associate's degree             0.106
        high school                    0.102
female  high school                    0.094
        some high school               0.091
male    some high school               0.088
female  bachelor's degree              0.063
male    bachelor's degree              0.055
female  master's degree                0.036
male    master's degree                0.023
dtype: float64

It is found that there is a 5.9% students parental level of education is masters degree,11.8% account of bachelors degree,22.2% accounts associate degree,22.6% some college,17.9% some high school and 19.6% high school.It is observed that more than 90% of students parents level of education is under graduate.

# 3) WHO SCORES THE MOST ON AVERAGE FOR MATH,READING AND WRITING BASED ON GENDER, TEST PREPARATION COURSE

In [10]:
gender_female=data['gender']=='female'
data[gender_female].head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [11]:
data.loc[data['gender']=='female','math score'].mean()

63.37644787644788

In [12]:
data.loc[data['gender']=='male','math score'].mean()

68.82157676348548

In [13]:
data.loc[data['gender']=='female','reading score'].mean()

72.59073359073359

In [14]:
data.loc[data['gender']=='male','reading score'].mean()

65.54564315352697

In [15]:
data.loc[data['gender']=='female','writing score'].mean()

72.46718146718146

In [16]:
data.loc[data['gender']=='male','writing score'].mean()

63.446058091286304

1 Male students score most on average (68.2%) for maths test compared to female students(63.38%)
2)Female students score most on average (72.59%) for reading test compared to male students(65.55%)
3)2)Female students score most on average (72.46%) for writing test compared to male students(63.44%)

In [17]:
data['test preparation course'].value_counts()

none         642
completed    358
Name: test preparation course, dtype: int64

In [18]:
data.loc[data['test preparation course']=='none','math score'].mean()

63.78971962616822

In [19]:
data.loc[data['test preparation course']=='completed','math score'].mean()

69.96648044692738

In [20]:
data.loc[data['test preparation course']=='none','reading score'].mean()

66.41744548286604

In [21]:
data.loc[data['test preparation course']=='completed','reading score'].mean()

74.17597765363128

In [22]:
data.loc[data['test preparation course']=='none','writing score'].mean()

64.45794392523365

In [23]:
data.loc[data['test preparation course']=='completed','writing score'].mean()

74.68435754189944

1  students who completed participation course score most on average (69.97%) for maths test compared to the students(63.79%) who not completed the course.
2) students who completed participation course score most on average (74.18%) for reading test compared to the students(66.42%) who not completed the course.
3)students who completed participation course score most on average (74.68%) for reading test compared to the students(64.46%) who not completed the course.

# 4)WHAT DO YOU THINK ABOUT THE SCORING VARIATIONS FOR MATH, READING AND WRITING BASED ON GENDER AND TEST PREPARATION COURSE

In [24]:
data.loc[data['gender']=='female','math score'].var()

256.9585931607201

In [25]:
data.loc[data['gender']=='male','math score'].var()

211.88909688494755

In [26]:
data.loc[data['gender']=='female','reading score'].var()

207.6774381455231

In [27]:
data.loc[data['gender']=='male','reading score'].var()

200.21101008445382

In [28]:
data.loc[data['gender']=='female','writing score'].var()

220.36932705017793

In [29]:
data.loc[data['gender']=='male','writing score'].var()

202.41392413799062

1) The scoring variation in maths test of female students( variation is 256.95) is higher than that of male students( 211.89% variation)
2)The scoring variation in reading test of female students( variation is 207.68) is higher than that of male students( 200.21% variation)
3)The scoring variation in writing test of female students( variation is 220.36) is higher than that of male students( 202.41% variation)


In [30]:
data.loc[data['test preparation course']=='none','math score'].var()

246.66866169973943

In [31]:
data.loc[data['test preparation course']=='completed','math score'].var()

210.8840273539583

In [32]:
data.loc[data['test preparation course']=='none','reading score'].var()

213.41985118657084

In [33]:
data.loc[data['test preparation course']=='completed','reading score'].var()

183.2658638874545

In [34]:
data.loc[data['test preparation course']=='none','writing score'].var()

226.25173866767676

In [35]:
data.loc[data['test preparation course']=='completed','writing score'].var()

175.20261177096555

1)The scoring variation in maths test of students who completed the preparation course( variation is 210.88) is less than that of students who not completed the course( 246.66% variation)
2)The scoring variation in maths test of students who completed the preparation course( variation is 183.27) is less than that of students who not completed the course( 213.42% variation)
3)The scoring variation in maths test of students who completed the preparation course( variation is 175.2) is less than that of students who not completed the course( 226.25% variation)

# 5) TOP 25% OF STUDENTS BASED ON THEIR MATHS SCORE

In [43]:
x=data['math score'].count()
y= int (25/100*x)

In [44]:
stud_max=data.nlargest(y,['math score'])
stud_max[['gender','math score','race/ethnicity']]

gender  math score race/ethnicity
515  female         100        group E
516  female         100        group E
517  female         100        group E
996    male         100        group A
997    male         100        group D
..      ...         ...            ...
428  female          77        group D
429  female          77        group E
849    male          77        group A
850    male          77        group C
851    male          77        group D

[250 rows x 3 columns]

The above columns list top 25% students based on their maths score.